packages/@aws-cdk/aws-apigateway/README.md

```diff
@@ -146,6 +146,67 @@ const proxy = resource.addProxy({
 });
 ```
 
+### Models
+
+You may define models to provide structure to request payloads sent to a method,
+and/or payloads returned to a client from a method. A model may be added to a
+REST API as follows:
+
+```ts
+// Direct addition to REST API
+const beerModel = api.addModel('Beer', 'application/json', {
+    $schema: 'http://json-schema.org/draft-04/schema#',
+    title: 'Beer',
+    type: 'object',
+    properties: {
+      name: { type: 'string' },
+      style: { type: 'string' },
+      abv: { type: 'number' },
+      ibu: { type: 'number' },
+    },
+  },
+  'Simple model for defining a beer.');
+
+// Separate instantiation
+const breweryModel = new apiGateway.Model(this, 'breweryModel', {
+  restApi: api,
+  name: 'Brewery',
+  description: 'Simple model for defining a brewery.',
+  contentType: 'application/json',
+  schema: {
+    $schema: 'http://json-schema.org/draft-04/schema#',
+    title: 'Brewery',
+    type: 'object',
+    properties: {
+      name: { type: 'string' },
+      address: { type: 'string' },
+      // Reference another model
+      beers: { type: 'array', items: { $ref: beerModel.referenceForSchema } }
+    },
+  },
+});
+```
+
+Given a model definition, you may either define a binding to request or response of a method via:
+
+#### Request
+
+RequestModel support not yet implemented.
+
+#### Response
+
+```ts
+const method = api.beer.addMethod('GET', getBeerLambdaHandler, {
+  methodResponses: [{
+      statusCode: '200',
+      responseModels: {
```

Comment:
the model seems to know what content type it supports. Is there a more concise api that can take advantage of this (something like “addResponseModel”?

Reply:
Yeah, that seems like a useful way to help steer people away from accidental content type mismatches.